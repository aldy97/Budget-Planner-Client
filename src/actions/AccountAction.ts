export const UPDATE_BUDGET = "update_budget";
export const UPDATE_BUDGET_THRESHOLD = "update_budget_threshold";

export interface UpdateBudget {
  type: typeof UPDATE_BUDGET;
  budget: number;
}

export interface UpdateBudgetThreshold {
  type: typeof UPDATE_BUDGET_THRESHOLD;
  threshold: number;
}

export type AccountAction = UpdateBudget | UpdateBudgetThreshold;
