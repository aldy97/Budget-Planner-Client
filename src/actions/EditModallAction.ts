export const UPDATE_RECORD_ID = "update_record_id";
export const EDIT_TITLE = "edit_title";
export const EDIT_RECORD_DATE = "edit_record_date";
export const EDIT_RECORD_TYPE = "edit_record_type";
export const EDIT_CATEGORY = "edit_category";
export const EDIT_AMOUNT = "edit_amount";
export const EDIT_DESCRIPTION = "edit_description";
export const CLEAR_STATUS = "clear_status";

export interface UpdateRecordID {
  type: typeof UPDATE_RECORD_ID;
  recordID: string;
}

export interface EditTitle {
  type: typeof EDIT_TITLE;
  title: string;
}

export interface EditRecordDate {
  type: typeof EDIT_RECORD_DATE;
  date: string;
}

export interface EditRecordType {
  type: typeof EDIT_RECORD_TYPE;
  recordType: string;
}

export interface EditCategory {
  type: typeof EDIT_CATEGORY;
  category: string;
}

export interface EditAmount {
  type: typeof EDIT_AMOUNT;
  amount: number;
}

export interface EditDescription {
  type: typeof EDIT_DESCRIPTION;
  description: string;
}

export interface ClearStatus {
  type: typeof CLEAR_STATUS;
}

export type EditModalAction =
  | UpdateRecordID
  | EditTitle
  | EditRecordDate
  | EditRecordType
  | EditAmount
  | EditCategory
  | EditDescription
  | ClearStatus;
