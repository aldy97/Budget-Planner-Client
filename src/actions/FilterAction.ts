import { Filter } from "../reducers/FilterReducer";
export const UPDATE_FILTER = "update_filter";

export interface UpdateFilter {
  type: typeof UPDATE_FILTER;
  filter: Filter;
}

export type FilterAction = UpdateFilter;
