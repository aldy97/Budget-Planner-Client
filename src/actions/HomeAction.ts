import { User } from "../reducers/HomeReducer";
import { Record } from "../components/Overview/Content";

export const UPDATE_USER_INFO = "update user information";
export const UPDATE_RECORDS = "update records";

export interface UpdateUserInfo {
  type: typeof UPDATE_USER_INFO;
  user: User;
}

export interface UpdateRecords {
  type: typeof UPDATE_RECORDS;
  records: Record[];
}

export type HomeAction = UpdateUserInfo | UpdateRecords;
