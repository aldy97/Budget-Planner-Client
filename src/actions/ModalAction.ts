export const SWITCH_TO_UPDATE = "switch_to_update";
export const UPDATE_TITLE = "update_title";
export const UPDATE_RECORD_DATE = "update_record_date";
export const UPDATE_RECORD_TYPE = "update_record_type";
export const UPDATE_CATEGORY = "update_category";
export const UPDATE_AMOUNT = "update_amount";
export const UPDATE_DESCRIPTION = "update_description";
export const CLEAR_RECORD = "clear_record";

// this action is used when a record needs to be updated
export interface SwitchToUpdate {
  type: typeof SWITCH_TO_UPDATE;
  update: boolean;
}

export interface UpdateTitle {
  type: typeof UPDATE_TITLE;
  title: string;
}

export interface UpdateRecordDate {
  type: typeof UPDATE_RECORD_DATE;
  recordDate: string;
}

export interface UpdateType {
  type: typeof UPDATE_RECORD_TYPE;
  recordType: string;
}

export interface UpdateCategory {
  type: typeof UPDATE_CATEGORY;
  category: string;
}

export interface UpdateAmount {
  type: typeof UPDATE_AMOUNT;
  amount: number;
}

export interface UpdateDescription {
  type: typeof UPDATE_DESCRIPTION;
  description: string;
}

export interface ClearRecord {
  type: typeof CLEAR_RECORD;
}

export type ModalAction =
  | SwitchToUpdate
  | UpdateAmount
  | UpdateCategory
  | UpdateDescription
  | UpdateTitle
  | UpdateType
  | ClearRecord
  | UpdateRecordDate;
