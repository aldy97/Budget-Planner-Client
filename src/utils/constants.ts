export const COLORS = {
  THEMEBLUE: "#1890ff",
  UNSELECTEDOPTION: "#000000a6",
  ORANGE: "#ffa940",
};

export const ExpenseCategories = [
  "Commute",
  "Meals",
  "Shoppings",
  "Grocery",
  "Veggies",
  "Fruits",
  "Snacks",
  "Sports",
  "Entertainment",
  "Phone Plan",
  "Make up",
  "Housing",
  "Clothing",
  "Children",
  "Elders",
  "Social",
  "Travelling",
  "Wine",
  "Digitals",
  "Car",
  "Medical Care",
  "Books",
  "Learning",
  "Pets",
  "Gifts",
  "Office",
  "Maintainance",
  "Donation",
];

export const IncomeCategories = [
  "Salary",
  "Part-time",
  "Investment",
  "Gifts",
  "Others",
];

export const URL = "http://localhost:7001";
