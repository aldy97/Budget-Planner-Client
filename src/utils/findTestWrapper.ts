import { ReactWrapper } from "enzyme";

export const findTestWrapper = (
  wrapper: ReactWrapper,
  tag: string
): ReactWrapper => {
  return wrapper.find(`[data-test="${tag}"]`);
};
