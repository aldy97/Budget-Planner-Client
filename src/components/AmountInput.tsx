import React from "react";
import { InputNumber } from "antd";
import { UPDATE_AMOUNT, UpdateAmount } from "../actions/ModalAction";
import { connect } from "react-redux";
import { Dispatch } from "redux";

interface AmountInputProps {
  updateCategoryToRedux: (amount: number) => void;
}

function AmountInput({ updateCategoryToRedux }: AmountInputProps) {
  const onChange = (value: string | number | undefined): void => {
    updateCategoryToRedux(value as number);
  };

  return (
    <InputNumber
      style={{ display: "block" }}
      defaultValue={0}
      formatter={value => `$ ${value}`.replace(/\B(?=(\d{3})+(?!\d))/g, ",")}
      onChange={onChange}
      data-test="input"
    />
  );
}

const mapDispatch = (dispatch: Dispatch) => {
  return {
    updateCategoryToRedux(amount: number) {
      const action: UpdateAmount = {
        type: UPDATE_AMOUNT,
        amount,
      };
      dispatch(action);
    },
  };
};

export default connect(null, mapDispatch)(AmountInput);
