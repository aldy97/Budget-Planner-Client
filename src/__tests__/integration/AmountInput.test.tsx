import React from "react";
import { mount } from "enzyme";
import AmountInput from "../../components/AmountInput";
import { findTestWrapper } from "../../utils/findTestWrapper";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { UPDATE_AMOUNT, UpdateAmount } from "../../actions/ModalAction";

const mockStore = configureStore([]);

describe("Amount Input", () => {
  let store;
  let wrapper;
  let input;

  beforeEach(() => {
    store = mockStore({ ModalReducer: { amount: 200 } });

    store.dispatch = jest.fn();

    wrapper = mount(
      <Provider store={store}>
        <AmountInput></AmountInput>
      </Provider>
    );

    input = findTestWrapper(wrapper, "input").at(0);
  });

  it("renders without crashing", () => {
    expect(wrapper.length).toBe(1);
  });

  it("default value is 0", () => {
    expect(input.prop("defaultValue")).toBe(0);
  });

  it("renders default value", () => {
    const input = wrapper.find("input");
    expect(input.length).toBe(1);
    expect(input.prop("value")).toEqual("$ 0");
  });

  it("dispatches when amount value changes", () => {
    const input = wrapper.find("input");
    expect(store.dispatch).toHaveBeenCalledTimes(0);
    input.simulate("change", { value: 200 });
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    input.simulate("change", { value: 201 });
    expect(store.dispatch).toHaveBeenCalledTimes(2);
  });

  it("can dispatch action to the reducer", () => {
    const action: UpdateAmount = {
      type: UPDATE_AMOUNT,
      amount: 100,
    };
    store.dispatch(action);
    expect(store.getActions()).toMatchSnapshot();
  });
});
