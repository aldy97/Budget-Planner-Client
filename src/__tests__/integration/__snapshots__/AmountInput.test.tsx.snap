// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amount Input can dispatch action to the reducer 1`] = `Array []`;
